csrc/preseg_passes/remove_bcast_squeeze.cpp

```diff
@@ -362,18 +362,46 @@ TensorView* maybeDoReplacement(TensorView* orig) {
       replacement->toString(),
       " has different dimension than original ",
       orig->toString());
+  // Replacing `orig` with `replacement` can introduce resharding on the
+  // consumer of `orig`, e.g., the `cast` in the following example.
+  //
+  // [bDID{1}, i0]: replacement
+  //      |
+  //      | squeeze
+  //      v
+  //     [i0]
```

Comment:
Just so I understand correctly, this actually is a resharding op, but it is not skipped here because we don't check SqueezeOp or BroadcastOp for resharding. Is that right?

Reply:
> this actually is a resharding op

It's arguable. I tend to think it's not resharding because, assuming same mesh, broadcast dimensions are size-1 and they are replicated regardless of whether DID or not. 

Note: remove_bcast_squeeze correctly checks for isResharding when picking candidates: https://github.com/NVIDIA/Fuser/blob/63cc45c47f1e96946cb952e3923bd076867a8015/csrc/preseg_passes/remove_bcast_squeeze.cpp#L85-L87. But, for the reason that I said, https://github.com/NVIDIA/Fuser/blob/63cc45c47f1e96946cb952e3923bd076867a8015/csrc/multidevice/utils.cpp#L162-L165 doesn't treat as resharding a removal or a creation of a broadcast dimension.